### ISSUE TITLE
[Bug] Audio feedback breaks Bluetooth A2DP-to-HFP switch

### ISSUE THREAD

**Issue by oranellis** [peteonrails/voxtype#773]
## Foreword

Hi, I have used gpt-5.6-sol to assist in diagnosing and generating this bug report, I have manually reviewed a lot of the behavior and validated the findings with testing so I suspect this issue is as described by the model.

## Description

When AirPods are the default input and output, WirePlumber normally switches them from A2DP to HFP while Voxtype records, then restores A2DP afterward. With Voxtype audio feedback enabled, the start cue coincides with the profile switch, Bluetooth transports enter an error state, and recording aborts. Disabling only audio feedback fixes it.

The daemon does not actually crash: it remains active with `NRestarts=0` and produces no coredump.

## Steps to Reproduce

1. Connect AirPods as the default input/output while using A2DP.
2. Ensure WirePlumber has `bluetooth.autoswitch-to-headset-profile = true`.
3. Run:

   ```sh
   voxtype config set audio.feedback.enabled true
   systemctl --user restart voxtype.service
   voxtype record start
   ```

4. The OSD appears briefly, transport errors occur, and recording aborts.
5. Repeat after setting `audio.feedback.enabled false`: recording succeeds, HFP capture becomes active, and A2DP is restored afterward.

## Expected Behavior

Play the start/stop cues while temporarily using HFP for capture, then return to A2DP.

## Relevant Logs

```text
voxtype_osd_gtk4: showing ...
wireplumber: pw.node: (bluez_output.<redacted>-89) running -> error
wireplumber: spa.bluez5: Failure in Bluetooth audio transport .../sep2/fd0
wireplumber: bluez_input.<redacted>.0 suspended -> error
voxtype_osd_gtk4: hiding ...
Bluetooth: hci0: SCO packet for unknown connection handle 5
```

## Environment

- Arch Linux, kernel `7.2.6.arch2-1`
- Hyprland `0.56.2` under UWSM
- Voxtype/`voxtype-bin` `1.0.1`, ONNX AVX2, Cohere engine
- PipeWire `1.6.8`, WirePlumber `0.5.17`, BlueZ `5.87`
- AirPods Pro; Qualcomm-based USB Bluetooth adapter `0489:e0a2`
- Available profiles: A2DP/AAC and HFP with mSBC or CVSD

## Diagnostic Results

| Test | Result |
|---|---|
| Voxtype, feedback enabled | Fails during profile switch |
| Voxtype, feedback disabled | Works and restores A2DP |
| Direct `pw-record` with automatic switch | Works and restores A2DP |
| Manual mSBC or CVSD capture | Works |

## Suspected Cause

`AudioFeedback::new()` retains a Rodio `OutputStream` opened on the default output, and `play_wav()` starts cues asynchronously with `sink.detach()`. During capture startup, A2DP nodes disappear and are replaced by HFP nodes. The retained feedback handle may therefore target the disappearing A2DP node.

Reopening feedback output after capture/profile readiness—or otherwise synchronizing cue playback with that transition—would be more robust than a fixed delay. On stop, HFP should remain open until the stop cue finishes before capture is closed.

I searched all open and closed issues via GitHub's API on 2026-09-17 and found no duplicate. The closest are [#100](https://github.com/peteonrails/voxtype/issues/100) (PipeWire source selection), [#651](https://github.com/peteonrails/voxtype/issues/651) (capture-ready signalling), and [#642](https://github.com/peteonrails/voxtype/issues/642) (mid-recording device disconnect); none describes this feedback-dependent profile-switch failure.
